hack/netpol/pkg/main/main.go

```diff
@@ -946,3 +973,102 @@ func testPodLabelWhitelistingFromBToA() []*TestStep {
 		},
 	}
 }
+
+// This covers tests with two ingress policies with CIDRs.
+// Policy-1 with FromCIDR and ExceptCIDR and Policy-2 with FromCIDR that is same as policy-1 ExceptCIDR
+// 1) Enforce policy-1 by checking no pod is able to communicate pod from Except block
+// 2) Enforce policy-1 and policy-2 by checking all pods are able to communicate the pod on Except block in policy-1
+// For simplicity picking FromCIDR and ExceptCIDR to be same, and this is the podCIDR of node in which pod "y/a" is present.
+func testIngressOverlapCIDRBlocks() []*TestStep {
+	exceptBuilder1 := make([]string, 1)
+	exceptBuilder3 := make([]string, 1)
+
+	// It does not matter if podCIDR of pod "x/a" and "y/a" are same
+	allowCIDRStr := podCIDRs["y/a"].String()
+	exceptBuilder1[0] = podCIDRs["y/a"].String()
+	// Build exceptCIDR to block just the pod "y/a"
+	exceptBuilder3[0] = podIPs["y/a"] + "/32"
+
+	exceptAndIsolated := func() *Reachability {
+		r := NewReachability(allPods, true)
+		r.ExpectAllIngress(Pod("x/a"), false)
+		r.Expect(Pod("x/a"), Pod("x/a"), true)
+		return r
+	}
+
+	allowAndIsolated := func() *Reachability {
+		r := NewReachability(allPods, true)
+		r.ExpectAllIngress(Pod("x/a"), false)
+		r.Expect(Pod("x/a"), Pod("x/a"), true)
+		// Adding connectivity for all the pods in node of pod "y/a"
```

Comment:
the idea of IPBlock in general is to allow traffic external to cluster.. we can try to mimic that by taking up @antoninbas suggestion?

Reply:
or we can simplify this by checking if PodIP falls under the allowCIDR built by you(instead of using node CIDRs) and then set the expectation as true or false?